File: app/src/main/java/com/esoxjem/movieguide/listing/ListingComponent.java
package com.esoxjem.movieguide.listing;

import dagger.Subcomponent;

/**
 * @author arunsasidharan
 */
@ListingScope
@Subcomponent(modules = {ListingModule.class})
public interface ListingComponent
{
    MoviesListingFragment inject(MoviesListingFragment fragment);
}

File: app/src/main/java/com/esoxjem/movieguide/movies/listing/ListingComponent.kt
package com.esoxjem.movieguide.movies.listing

import com.esoxjem.movieguide.movies.sorting.SortingDialogFragment
import com.esoxjem.movieguide.movies.sorting.SortingModule

import dagger.Subcomponent

/**
 * @author arunsasidharan
 */
@ListingScope
@Subcomponent(modules = [(ListingModule::class), (SortingModule::class)])
interface ListingComponent {
    fun inject(fragment: MoviesListingFragment): MoviesListingFragment

    fun inject(fragment: SortingDialogFragment): SortingDialogFragment
}
